### DISCUSSION TITLE
how to make TriggerDAGRunOperator to just wait if run already exists?

### DISCUSSION THREAD

**Discussion by kalyanr23** [apache/airflow#29529] [Q&A]
So, I have two DAG's  `dag_1` and `dag_2`. I want to trigger `dag_2` in a task of `dag_1` and wait till `dag_2` is complete. But, sometimes, i clear and rerun `dag_1`. Since, the `dag_2` run already exists, its throwing `DagRunAlreadyExists` as expected. But, is there a way to just wait on completion of `dag_2` run if it already exists?

**Comment by potiuk** [comment]
```
:param reset_dag_run: Whether clear existing dag run if already exists.
        This is useful when backfill or rerun an existing dag run.
        This only resets (not recreates) the dag run.
        Dag run conf is immutable and will not be reset on rerun of an existing dag run.
        When reset_dag_run=False and dag run exists, DagRunAlreadyExists will be raised.
        When reset_dag_run=True and dag run exists, existing dag run will be cleared to rerun.
```

**Reply by kalyanr23** [reply]
@potiuk , I'm suggesting a change here.. please let me know if this can be considered.
can we add an argument which will make the task just wait on the Already Existing DAG run?

**Reply by potiuk** [reply]
Sure. You can always make a PR for that - and it's up to those who will approve (or not) a PR if you create one.

**Reply by potiuk** [reply]
This is **small** thing so *just doing it* (and showing off the code that does it) is far faster than asking if it can be done.

**Comment by raphaelauv** [comment]
https://github.com/apache/airflow/pull/39173 will be part of airflow 2.10
